plugins/by-name/fidget/default.nix

```diff
@@ -34,650 +28,424 @@ let
 
-        ignore = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-          List of LSP servers to ignore.
+        format_annote = defaultNullOpts.mkRaw "function(msg) return msg.title end" ''
+          How to format a progress annotation.
         '';
 
-        # Options related to how LSP progress messages are displayed as notifications
-        display = {
-          renderLimit = helpers.defaultNullOpts.mkNullable (with types; either (enum [ false ]) number) 16 ''
-            How many LSP messages to show at once.
-
-            If `false`, no limit.
-
-            This is used to configure each LSP notification group, so by default, this is a
-            per-server limit.
-          '';
-
-          doneTtl = helpers.defaultNullOpts.mkStrLuaOr types.ints.unsigned "3" ''
-            How long a message should persist after completion.
-
-            Set to `0` to use notification group config default, and `math.huge` to show
-            notification indefinitely (until overwritten).
+        format_group_name = defaultNullOpts.mkRaw "function(group) tostring(group) end" ''
+          How to format a progress notification's group name.
+        '';
 
-            Measured in seconds.
-          '';
+        overrides =
+          defaultNullOpts.mkAttrsOf notificationConfigType
+            {
+              rust_analyzer = {
+                name = "rust-analyzer";
+              };
+            }
+            ''
+              Override options from the default notification config.
+              Keys of the table are each notification group's `key`.
+            '';
+      };
 
-          doneIcon = mkIconOption "✔" "Icon shown when all LSP progress tasks are complete.";
+      lsp = {
+        progress_ringbuf_size = defaultNullOpts.mkNum 0 ''
+          Configure the nvim's LSP progress ring buffer size.
+        '';
 
-          doneStyle = helpers.defaultNullOpts.mkStr "Constant" ''
-            Highlight group for completed LSP tasks.
-          '';
+        log_handler = defaultNullOpts.mkBool false ''
+          Log `$/progress` handler invocations (for debugging).
+        '';
+      };
+    };
 
-          progressTtl = helpers.defaultNullOpts.mkStrLuaFnOr types.ints.unsigned "math.huge" ''
-            How long a message should persist when in progress.
-          '';
+    notification = {
+      poll_rate = defaultNullOpts.mkNum 10 ''
+        How frequently to update and render notifications.
+        Measured in Hertz (frames per second).
+      '';
 
-          progressIcon = mkIconOption ''
-            {
-              pattern = "dots";
-            }
-          '' "Icon shown when LSP progress tasks are in progress";
+      filter = defaultNullOpts.mkLogLevel "info" ''
+        Minimum log level to display.
+      '';
 
-          progressStyle = helpers.defaultNullOpts.mkStr "WarningMsg" ''
-            Highlight group for in-progress LSP tasks.
-          '';
+      history_size = defaultNullOpts.mkNum 128 ''
```

Comment:
This doesn't seem like it should be changed, since a message count is integer.

Reply:
Just mirrored what it is in: https://github.com/j-hui/fidget.nvim/blob/9238947645ce17d96f30842e61ba81147185b657/lua/fidget/notification.lua#L211

Reply:
That's fine, then. I'd rather track their upstream usage than artificially limit someone from a config upstream supports.